sklearn/gaussian_process/tests/test_gpc.py

```diff
@@ -0,0 +1,78 @@
+"""Testing for Gaussian process classification """
+
+# Author: Jan Hendrik Metzen <jhm@informatik.uni-bremen.de>
+# Licence: BSD 3 clause
+
+from copy import deepcopy
+
+import numpy as np
+
+from scipy.optimize import approx_fprime
+
+from sklearn.gaussian_process import GaussianProcessClassification
+from sklearn.gaussian_process.kernels import RBF
+
+from sklearn.utils.testing import (assert_true, assert_greater,
+                                   assert_almost_equal, assert_array_equal)
+
+
+f = lambda x: x * np.sin(x)
+X = np.atleast_2d(np.linspace(0, 10, 30)).T
+X2 = np.atleast_2d([2., 4., 5.5, 6.5, 7.5]).T
+y = np.array(f(X).ravel() > 0, dtype=int)
+
+
+kernels = [RBF(0.1), RBF((1e-3, 1.0, 1e3)),
+           (1e-2, 1.0, 1e2) * RBF((1e-3, 0.1, 1e3))]
+
+
+def test_predict_consistent():
+    """ Check binary predict decision has also predicted probability above 0.5.
+    """
+    for kernel in kernels:
+        kernel = deepcopy(kernel)
+        gpc = GaussianProcessClassification(kernel=kernel).fit(X, y)
+        assert_array_equal(gpc.predict(X),
+                           gpc.predict_proba(X) >=0.5)
+
+
+def test_lml_improving():
+    """ Test that hyperparameter-tuning improves log-marginal likelihood. """
```

Comment:
I believe it is more common to call the kernel's parameters  hyperparameters when speaking about the GP.